### ISSUE TITLE
uvu is not detecting a thrown string?

### ISSUE THREAD

**Issue by jasikpark** [lukeed/uvu#130]
In #128 there's a function call that is only detected by uvu if i catch and rethrow it, it just fails to complete the test suite otherwise.

**Comment by jasikpark** [comment]
I'm not sure that it's a thrown string that is being missed, but if you clone the astro-prettier-plugin repo, and run `yarn test` the first test "basic" is the culprit. I'll see if I can print out more diagnostics for why it's not detected when I get back to my laptop

**Comment by jasikpark** [comment]
minimal reproduction of the bug:
https://codesandbox.io/s/gallant-lederberg-9rmoy?file=/package.json:158-171 open the link and open a new terminal and run `yarn test`.

**Comment by jasikpark** [comment]
ah, this is a duplicate of https://github.com/lukeed/uvu/issues/60

**Comment by lukeed** [comment]
It wouldnt be the same cause as 60. Your CSB doesn't work btw, 502 bad gateway

**Comment by jasikpark** [comment]
> It wouldnt be the same cause as 60. Your CSB doesn't work btw, 502 bad gateway

yeah, i don't think that was the right way to write my reproduction - you need to open a new terminal and run 'yarn test' to see the output.

**Comment by jasikpark** [comment]
Here's a better link for the reproduction: https://playground.denobr.com/801f4d5301

**Comment by lukeed** [comment]
Think you forgot to save 😆 – that's just a "hello world" for a Deno playground.

No need to fiddle with this. Saving your CSB file contents here & will repro locally soon:

```js
import { test } from "uvu";

test("this test should fail", () => {
  console.log("before throw of string");
  throw "not detected by uvu";
  console.log("after throw of string");
  throw Error("hmmmm");
});

test.run();
```

**Comment by jasikpark** [comment]
basically the console.logs are meant to demonstrate where the async function stops executing

**Comment by lukeed** [comment]
Yup, understood.

**Comment by jasikpark** [comment]
The error is caused by `format()` accessing nonexistent properties on the string, causing an uncaught error.

**Comment by jasikpark** [comment]
```js
function format(name, err, suite = '') {
	let { details, operator='' } = err;
	let idx = err.stack && err.stack.indexOf('\n');
	if (err.name.startsWith('AssertionError') && !operator.includes('not')) details = compare(err.actual, err.expected); // TODO?
	let str = '  ' + FAILURE + (suite ? kleur.red(SUITE(` ${suite} `)) : '') + ' ' + QUOTE + kleur.red().bold(name) + QUOTE;
	str += '\n    ' + err.message + (operator ? kleur.italic().dim(`  (${operator})`) : '') + '\n';
	if (details) str += GUTTER + details.split('\n').join(GUTTER);
	if (!!~idx) str += stack(err.stack, idx);
	return str + '\n';
}

```

when `err` is a string, it doesn't have a `err.name` property, and so it throws

**Comment by lukeed** [comment]
Yes I know. I'm keeping this open as an exploration to see if it even makes sense to throw non-Errors.

The entirety of uvu operates with Error primitives. That's how assertion control is orchestrated. Throwing things that are not Errors is not the same thing as throwing an Error. In other words, `throw` on its own is not exclusively a "this is an error" -- it's a way to abort and bubble up any value, kinda the same way `goto` works in other languages. Errors are Errors.

Tbh my inclination is to close this as unsupported, but going to play with it first.

**Comment by jasikpark** [comment]
I would say that if you end up not supporting detecting non-Error() throws, that should be added to the documentation with a recommendation for how to correctly catch and rethrow them for uvu. 

Because I agree that encouraging Error() use is a good idea, but I haven't encountered many examples of using `throw` for things other than error reporting, even if it's via a raw string - maybe it could catch other values and print a warning that they aren't true errors, but are strings, so they should be changed
